### ISSUE TITLE
Corner case issue with OASISAR1 and s_min

### ISSUE THREAD

**Issue by agiovann** [flatironinstitute/CaImAn#791]
For better support, please use the template below to submit your issue. When your issue gets resolved please remember to close it.

Sometimes errors while running CNMF occur during parallel processing which prevents the log to provide a meaningful error message. Please reproduce your error with setting `dview=None`.

If you need to upgrade CaImAn follow the instructions given in the [documentation](https://caiman.readthedocs.io/en/master/Installation.html#upgrading).

- **Tell us a bit about your setup**: 
 1. Operating system (Linux/macOS/Windows): 
 Linux/macOS
 2. Python version (3.x): 
 3.7
 3. Working environment (Python IDE/Jupyter Notebook/other):
 Spyder
 4. Which of the demo scripts you're using for your analysis (if applicable):
```
#%%
import matplotlib.pyplot as plt
import numpy as np
from caiman.source_extraction.cnmf.temporal import constrained_foopsi
from scipy.interpolate import interp1d

#%%
def gen_data(trueSpikes=None, g=[.95], g_noise=0.01, sn=.3, poiss_noise_factor=1.5, T=3000, framerate=30, 
             firerate=.5, b=0, N=20, seed=None, nonlinearity=False, spike_noise=0.1):
    """
    Generate data from homogenous Poisson Process

    Parameters
    ----------
    trueSpikes: array, optional , default None
        input spike train
    g : array, shape (p,), optional, default=[.95]
        Parameter(s) of the AR(p) process that models the fluorescence impulse response.
    sn : float, optional, default .3
        Noise standard deviation.
    poiss_noise_factor: float
        factor multiplying the poisson noise (computed over the range of the data), typical values, 2-4
    g_noise:  float 
        factor multiplying noise on g
    T : int, optional, default 3000
        Duration.
    framerate : int, optional, default 30
        Frame rate.
    firerate : int, optional, default .5
        Neural firing rate.
    b : int, optional, default 0
        Baseline.
    N : int, optional, default 20
        Number of generated traces.
    seed : int, optional, default 13
        Seed of random number generator.    
    nonlinearity: boolean, default False
        whether add nonlinearity or not
    spike_noise: float
        noise added to the spike    

    Returns
    -------
    y : array, shape (T,)
        Noisy fluorescence data.
    c : array, shape (T,)
        Calcium traces (without sn).
    s : array, shape (T,)
        Spike trains.
    g : ground truth autoregressive factors
    """
    
    if trueSpikes is None:
        np.random.seed(seed)
        Y = np.zeros((N, T))
        trueSpikes = np.random.rand(N, T) < firerate / float(framerate) 
    else:
        N, T = trueSpikes.shape
        Y = np.zeros((N, T))
    
    truth = trueSpikes.astype(float)
    truth[truth>0] += np.random.normal(0,spike_noise,truth[truth>0].shape)
#    truth += np.random.normal(0,spike_noise,truth.shape)
    g = np.vstack(g*N)
    g += g_noise*2*(np.random.random(g.shape)-0.5)
    
    for i in range(2, T):
        if np.shape(g)[-1] == 2:            
            raise Exception('Not compatible with other changes')
            truth[:, i] += g[:,0].T * truth[:, i - 1] + g[:,1].T * truth[:, i - 2]
        else:
            truth[:, i] += g[:,0] * truth[:, i - 1]
            
    if nonlinearity:
        f, inv_f = nonlinear_transformation()
        truth = f(truth)
    else:
        inv_f = None
#        truth = truth / truth.max(axis=1)[:, np.newaxis] * maximum[:, np.newaxis]
    fl_sig = b+truth
    fl_sig = (fl_sig-fl_sig.min())/(fl_sig.max()-fl_sig.min())
    poiss_noise = np.random.poisson(fl_sig*1000)/1000*poiss_noise_factor
    Y = b + truth + sn * np.random.randn(N, T) + poiss_noise           
    return Y, truth, trueSpikes, g
    
def nonlinear_transformation(model='dana_kim_gc6f'):
    """
    Output function for nonlinear transformation. The transformation is based on
    the property of GCAMP6f
    
    Args:
    ----     
    model: str
        which model to use: 'dana_kim_gc6f', 'druckman_gc6f-TG'
       
    Returns:
    -------
    f : function
        Nonlinear transformation
    """
    x = np.array(range(-2,160))
    if model == 'dana_kim_gc6f':
        x_orig = np.array([-2, 0.0, 1.0, 2.0,  3.0, 5.0,   10,  20, 40 , 160])
        y_orig = np.array([-0.01, 0.0, 0.1,0.25, 0.5,0.9, 1.85,3.6, 4.9, 6.1])    
    elif 'druckman_gc6f-TG': # for extrapolation used values from dana kim
        x_orig = np.array([-2,     0.0, 1.0, 2.0, 3.0,  4.0, 10,  20, 40 , 160])
        y_orig = np.array([-0.01,  0.0, 0.12, 0.3, 0.62, 1.1, 1.85, 3.6, 4.9, 6.1])    
         
    f = interp1d(x_orig, y_orig, kind='quadratic', fill_value='extrapolate')
    inv_f = interp1d(y_orig, x_orig, kind='quadratic', fill_value='extrapolate')
    # would be better to fit to a logistic
    xnew = np.arange(-2, 160, 0.01)
    ynew = f(xnew)
    plt.figure()
    plt.title('Nonlinear transformation')
    plt.plot(xnew, ynew)
    plt.plot(x_orig, y_orig,'o')
    plt.xlabel('# AP')
    plt.ylabel('deltaF/F')
    return f, inv_f
#%%
Y, truth, trueSpikes, gs = gen_data(firerate=2, N=1000, sn=0.1 ,poiss_noise_factor=0)
for index in range(len(Y)):
    c, bl, c1, g, sn, s, lam = constrained_foopsi(Y[index], p=1, s_min=0.01)
    idx_bad = np.where((s>0) & (s<0.01))[0]
    if len(idx_bad)>0:
        print(idx_bad)
        plt.plot(Y[index])
        plt.plot(idx_bad,1,'*')
        plt.xlim([0,100])
        plt.pause(1)
        plt.cla()
```
 5. CaImAn version*: 
 latest on dev branch
 6. CaImAn installation process (`pip install .`/`pip install -e .`/conda): 
pip install -e .
*You can get the CaImAn version by creating a `params` object and then typing `params.data['caiman_version']`. If the field doesn't exist, type N/A and consider upgrading)

- **Describe the issue that you are experiencing**
When setting s_min to a value larger than 0, there is a problematic  corner case: sometimes the first transient has value for spikes that is is 0<s<s_min, so not fulfilling the constraint s>s_min

- **Copy error log below**

- **If you're not reporting an error, type your message below**

**Comment by j-friedrich** [comment]
This could happen as aftereffect when the interim estimate of the initial calcium `c1` was negative and is set to `0` in the last step. 
Fixed it in the dev branch.

**Comment by j-friedrich** [comment]
Solved in the recent 1.8.5 release.
